Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -492,6 +492,15 @@ protected virtual async Task<CallInvoker> CreateCallInvokerAsync(CancellationTok
             return channel.CreateCallInvoker();
         }
 
+        /// <summary>
+        /// Obtains the service endpoint taking into account <see cref="Endpoint"/> and <see cref="UniverseDomain"/>.
+        /// Override this method in a concrete builder type if an endpoint may be customized further.
+        /// </summary>
+        protected virtual string GetEffectiveEndpoint() => Endpoint ??
```

Comment:
Is there a reason why this is a method whereas EffectiveUniverseDomain is a property?

Reply:
A property could still be overridden... CanUseChannelPool is virtual, for example.

Reply:
Yep, but for some reason I though we were leaving properties non-virtual and having getter methods for the overridable aspects. Didn't look closely at CanUseChannelPool.

I'll make this one a property then.